lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -418,17 +418,39 @@ public function commit($entity = null)
 
         $this->dispatchPostFlushEvent();
 
+        $this->postCommitClear($entity);
+    }
+
+    /**
+     * @param null|object|array $entity
+     */
+    private function postCommitClear($entity = null) {
+
         // Clear up
         $this->entityInsertions =
         $this->entityUpdates =
         $this->entityDeletions =
         $this->extraUpdates =
-        $this->entityChangeSets =
         $this->collectionUpdates =
         $this->collectionDeletions =
         $this->visitedCollections =
-        $this->scheduledForSynchronization =
         $this->orphanRemovals = array();
+
+        if ($entity === null) {
```

Comment:
Sorry but I don't understand what you want. I used the exact same code as the one of the beginning of the commit method.